src/commands/test/test_environment_exceptions.py

```diff
@@ -73,6 +76,14 @@ def __str__(self) -> str:
 
         return "\n".join(result)
 
+    def __reduce__(self):
+        return type(self), (
+            self.error_message,
+            self.error_type,
+            self.code,
+            self.details,
+        )
+
```

Comment:
Why do we need this?

Reply:
for [pickling](https://docs.python.org/3/library/pickle.html)

Reply:
Isn't pickling supported by default?

Reply:
> In most cases, no additional code is needed to make instances picklable. By default, pickle will retrieve the class and the attributes of an instance via introspection.

https://docs.python.org/3/library/pickle.html#pickling-class-instances

Reply:
There were some issues because with how we inherit from error class. We would have to make arguments optional. I somewhat prefer this explicit aproach. We can revisit it later, but seems not very painful/hacky for now.